### ISSUE TITLE
9.1.4.10 Inhalte brechen um: Breite 320 px nicht mehr mit Web Developer messbar

### ISSUE THREAD

**Issue by johannesFischer84** [BIK-BITV/BIK-Web-Test#57]
In der Prüfung heißt es noch, dass mit der Web Developer Toolbar über `Resize` -> `Edit Resize Dimensions...` die Breite auf 320 px eingestellt werden soll. Jedoch funktioniert dies nicht mehr, weil die Browser-Fensterbreite sich unabhängig von der Toolbar nicht mehr auf eine so kleine Breite reduzieren lässt.

Man kann es allerdings inzwischen über die Browser-Entwickler-Werkzeuge prüfen, wenn man dort das Symbol der mobilen Geräte (Firefox `Bildschirmgrößen testen`, Chrome `Toggle device toolbar`) auswählt. Daher wäre mein Vorschlag, den Punkt 2 der Prüfung so zu schreiben:
2. In den Browser-Entwickler-Werkzeugen das Symbol der mobilen Geräte (Firefox `Bildschirmgrößen testen`, Chrome `Toggle device toolbar`) auswählen und dort die Viewport-Breite auf 320 CSS-Pixel einstellen.

**Comment by detlevhfischer** [comment]
Eine Möglichkeit ist die Anzeige der devtools in der rechten Spalte. Damit lässt sich durch Verschieben der vertikalen Trennleiste die Breite des Browserfensters auf 320 px stellen (es gab auch in einem Browser die Möglichkeit der dynamischen Ausgabe der Breite, ich weiß nicht mehr in welchem, und wie einzustellen...). 
In meiner Erfahrung ist bei Mobilgeräten häufiger ein anderes Verhalten zu beobachten als bei der Vergrößerung auf 400% in einer Ausgangsbreite von 1280 px. Das kann auch an dem User agent string liegen. Ich weiß nicht, was bei Definition eines benutzerdefinierten Geräts hier zweckmäßigerweise stehen sollte. Ich weiß auch nicht, welchen Unterschied es macht, ob hier Der Haken bei "Touch.Gerät" gesetzt wir oder nicht (bei Firefox).
 Da Desktop mit starker Vergrößerung für sehbehinderte Nutzer wohl die relevantere Einstellung ist, wäre eine Möglichkeit, 1.4.10a grundsätzlich in Chrome 1280 px Breite und 400% zu prüfen. Das Problem ist dann, wie wir fest positionierte Bereiche bewerten, die im Hochformat ggf. kein Problem darstellen, im Querformat mit Vergrößerung aber oft keinen Raum mehr für Seiteninhalte lassen. Hier gibt es meines Wissens keine normative Festlegung in den WCAG (außer der advisory technique zum Lösen fest positionierter Bereiche bei einer engeren Viewport-Breite).
Entwicklungsbegleitend könnte man hier schon was abziehen, abschließend wohl nicht.

**Comment by johannesFischer84** [comment]
> Eine Möglichkeit ist die Anzeige der devtools in der rechten Spalte. Damit lässt sich durch Verschieben der vertikalen Trennleiste die Breite des Browserfensters auf 320 px stellen (im Chrome gibt es die dynamische Ausgabe der Breiteneinstellung oben rechts - bin mir nicht sicher, ob hier wirklich 320px einzustellen wären oder ggf. 336px um 16 px für den Scrollbalken abzuziehen).

Stimmt, die Verschiebung der Trennleiste ist eine weitere Möglichkeit, wobei ich die Anzeige der Pixel im Chrome noch nicht kannte. Danke für den Hinweis.  Ich habe es eben mal mit und ohne Scrollbalken ausprobiert und der Scrollbalken zählt tatsächlich in die Breite mit hinein. Bei 334 px in der Anzeige hatte ich die gleiche Breite ohne und mit Scrollbalken. Bei mir waren es also 14 px Unterschied.

> Das Problem ist dann, wie wir fest positionierte Bereiche bewerten, die im Hochformat ggf. kein Problem darstellen, im Querformat mit Vergrößerung aber oft keinen Raum mehr für Seiteninhalte lassen. Hier gibt es meines Wissens keine normative Festlegung in den WCAG (außer der advisory technique zum Lösen fest positionierter Bereiche bei einer engeren Viewport-Breite).

Ich kenne auch nur die vorgegebene Viewport-Breite, keine vorgegebene Höhe. Bei fixierten Bereichen kann man auch aus meiner Sicht nur den Tipp geben, am besten darauf zu verzichten (auch der Fokus interaktiver Elemente dahinter ist oft nicht erkennbar, wenn man nicht die Pfeiltasten zum Scrollen drückt). Bei der [Definition der Blitze in einer NOTE](https://www.w3.org/TR/WCAG/#h-note-83) ist in den WCAG mal eine Höhe von 768 px angegeben. Das könnte man ggf. als Hintertür bzw. als Argument für eine Aufteilung bei einer bestimmten Höhe verwenden.

**Comment by stefanFarnetani** [comment]
Spricht den etwas dagegen, wenn man die von @johannesFischer84 beschriebene Browser-Entwickler-Werkzeuge verwendet. Ich finde diese viel praktischer als die devtools.

**Comment by cgesting** [comment]
Hallo zusammen,

ich freue mich, an der Diskussion teilhaben zu dürfen. Ich bin zufällig über dieses Issue hier gestoßen.
Da wir ein IT-Dienstleister sind und die Welt sehr technisch sehen wollte ich hier noch den Tipp geben, sich auch mal Browserstack anzusehen: https://www.browserstack.com

Für manche neuen Features oder der Beseitigung von Bugs, welche lediglich in einer spezifischen Version eines Webbrowsers oder bei einem Webbrowser eines spezifischen Herstellers auftreten, kommt man nicht daran vorbei, die Änderungen in genau dieser Instanz zu testen. Da die lokale Installation von mehreren Browsern und insbesondere einer spezifischen Version aufwendig und nur schwierig möglich ist, bietet es sich an, auf externe Tools zurückzugreifen, welche genau für diesen spezifischen Anwendungsfall entwickelt wurden. Browserstack ist eine solche cloud-basierte Testplattform, um eine Webanwendungen auf Browsern unterschiedlicher Hersteller und unterschiedlicher Versionen testen zu können. Mittlerweile sind nahezu auch alle mobile Geräte verfügbar. Dabei ist es nicht nur möglich, bereits veröffentlichte Webanwendungen zu testen, sogar das Testen von lokalen Anwendungen wird über eine sichere Verbindung ermöglicht. Der vollständige Funktionsumfang von Browserstack umfasst noch weitere Produkte.

Ich finde das Tool echt beeindruckend und da wir als IT-Dienstleister in Besitz einer Lizenz sind, nutze ich das gerne punktuell immer mal wieder in verschiedenen Prüfschritten. Es steigert meiner Meinung erheblich die Testqualität.

Viel Spaß beim Ausprobieren!
